jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/internal/HttpChannelState.java

```diff
@@ -1157,37 +1151,34 @@ else if (last && !(totalWritten == 0 && HttpMethod.HEAD.is(_request.getMethod())
                     }
                 }
 
-                // If no failure by this point, we can try to send
+                // If no failure by this point, we can try to send.
                 if (failure == null)
                     failure = httpChannelState.lockedStreamSend(last, length);
 
                 // Have we failed in some way?
                 if (failure == DO_NOT_SEND)
                 {
                     httpChannelState._serializedInvoker.run(callback::succeeded);
+                    return;
```

Comment:
DO_NOT_SEND is not really a failure case, so it doesn't match the comment above.

Reply:
Renamed `DO_NOT_SEND` to `NOTHING_TO_SEND` and moved the comment.